shell.nix

```diff
@@ -1,67 +1,106 @@
-{ pkgs ? import <nixpkgs> {} }: with pkgs;
+{ pkgs ? import ./pinned.nix {} }:
 
-mkShell {
-  FONTCONFIG_FILE = makeFontsConf { fontDirectories = [ inconsolata-lgc libertine libertinus]; };
-  buildInputs = [
-    (texlive.combine {
-        inherit (texlive)
-        bookcover
-        textpos
-        fgruler
-        tcolorbox
-        fvextra
-        framed
-        newtx
-        nowidow
-        emptypage
-        wrapfig
-        subfigure
-        adjustbox
-        collectbox
-        tikz-cd
-        imakeidx
-        idxlayout
-        titlesec
-        subfiles
-        lettrine
-        upquote
-        libertine
-        mweights
-        fontaxes
-        mdframed
-        needspace
-        xifthen
-        ifnextok
-        currfile
-        noindentafter
-        ifmtarg
-        scheme-medium
-        listings
-        minted
-        microtype
-        babel
-        todonotes
-        chngcntr
-        ifplatform
-        xstring
-        minifp
-        titlecaps
-        enumitem
-        environ
-        trimspaces
-        l3packages;
-      })
+let
+  #############################################################################
+  # LaTeX Environment
+  texliveEnv = pkgs.texlive.combine {
+    inherit (pkgs.texlive)
+      bookcover
+      textpos
+      fgruler
+      tcolorbox
+      fvextra
+      framed
+      newtx
+      nowidow
+      emptypage
+      wrapfig
+      subfigure
+      adjustbox
+      collectbox
+      tikz-cd
+      imakeidx
+      idxlayout
+      titlesec
+      subfiles
+      lettrine
+      upquote
+      libertine
+      mweights
+      fontaxes
+      mdframed
+      needspace
+      xifthen
+      ifnextok
+      currfile
+      noindentafter
+      ifmtarg
+      scheme-medium
+      listings
+      minted
+      microtype
+      babel
+      todonotes
+      chngcntr
+      ifplatform
+      xstring
+      minifp
+      titlecaps
+      enumitem
+      environ
+      trimspaces
+      l3packages
+      zref
+      catchfile
+      import
+      ;
+  };
+
+  #############################################################################
+  # Python Environment
+
+  # Pin the Python version and its associated package set in a single place.
+  python = pkgs.python38;
+  pythonPkgs = pkgs.python38Packages;
```

Comment:
Adding these aliases here makes it a bit easier to update and/or modify Python versions in the future.